hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2469,6 +2472,24 @@ protected UniqueEntityLoader createEntityLoader(
 				.buildLoader( this, batchSize, lockOptions, getFactory(), loadQueryInfluencers );
```

Comment:
@sebersole ah, I misplaced the comments about the overriding method.

The issue is here: if the user has its own `AbstractEntityPersister` overriding the two `createEntityLoaders()` methods above, with this patch, they won't be called for most lock modes as it's the 2 methods below with the template that will be called.

That's what was bothering me. I don't know if it's a use case we support: I saw a documentation example with a custom class extending `SingleTableEntityPersister` somewhere so I suppose it can happen.

And this part would definitely breaks it.

Reply:
Indeed. It even overrides `createLoaders()` due to a TODO we should probably fix. I should have thought about OGM!